### PR TITLE
feat (charges): add invoiceable field

### PR DESCRIPTION
**Pull request by lovrocolic**
## Context

This feature adds possibility to bill charges in advance

## Description

In this PR new field called `invoiceable` is added. This field is responsible for selecting if certain charges will be included on invoice or not.

### CODE REVIEW

File: app/services/plans/update_service.rb [hunk 1]
```diff
@@ -87,10 +88,10 @@ def process_charges(plan, params_charges)
           unless plan.attached_to_subscriptions?
             payload_charge[:group_properties]&.map! { |gp| GroupProperty.new(gp) }
 
-            pay_in_advance = payload_charge.delete(:pay_in_advance)
+            invoiceable = payload_charge.delete(:invoiceable)
             min_amount_cents = payload_charge.delete(:min_amount_cents)
             if License.premium?
-              charge.pay_in_advance = pay_in_advance || false
+              charge.invoiceable = invoiceable unless invoiceable.nil?
```
Comment (jdenquin) [inline app/services/plans/update_service.rb:94]: Metrics/BlockNesting: Avoid more than 3 levels of block nesting. (https://rubystyle.guide#three-is-the-number-thou-shalt-count)

File: spec/services/plans/create_service_spec.rb [hunk 2]
```diff
@@ -117,8 +118,19 @@
...
         plan = plans_service.create(**create_args).plan
 
-        expect(plan.charges.standard.first).to have_attributes(pay_in_advance: false, min_amount_cents: 100)
-        expect(plan.charges.graduated.first).to be_pay_in_advance
+        expect(plan.charges.standard.first).to have_attributes(
+          {
+            pay_in_advance: false,
+            min_amount_cents: 100,
+            invoiceable: true,
+          }
```
Comment (jdenquin) [inline spec/services/plans/create_service_spec.rb:126]: Style/TrailingCommaInArguments: Put a comma after the last parameter of a multiline method call. (https://rubystyle.guide#no-trailing-params-comma)

File: spec/services/plans/create_service_spec.rb [hunk 3]
```diff
@@ -117,8 +118,19 @@
...
+        )
+        expect(plan.charges.graduated.first).to have_attributes(
+          {
+            pay_in_advance: true,
+            invoiceable: false,
+          }
```
Comment (jdenquin) [inline spec/services/plans/create_service_spec.rb:132]: Style/TrailingCommaInArguments: Put a comma after the last parameter of a multiline method call. (https://rubystyle.guide#no-trailing-params-comma)

File: app/services/plans/create_service.rb [hunk 4]
```diff
@@ -44,12 +44,13 @@ def create_charge(plan, args)
       charge = plan.charges.new(
         billable_metric_id: args[:billable_metric_id],
         charge_model: args[:charge_model]&.to_sym,
+        pay_in_advance: args[:pay_in_advance] || false,
```
Comment (rsempe) [inline app/services/plans/create_service.rb:47]: It's not premium anymore?

**Review by rsempe — APPROVED** [review]

**Pull request merged**